cfme/tests/automate/test_common_methods.py

```diff
@@ -180,3 +180,99 @@ def test_miq_password_decrypt(klass):
         execute_methods=True,
     )
     assert result.validate()
+
+
+@pytest.fixture(scope='module')
+def original_class(domain):
+    # Take the 'Request' class and copy it for own purpose.
+    domain.parent.instantiate(name="ManageIQ").namespaces.instantiate(
+        name="System"
+    ).classes.instantiate(name="Request").copy_to(domain.name)
+    klass = domain.namespaces.instantiate(name="System").classes.instantiate(name="Request")
+    yield klass
+    klass.delete_if_exists()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1700524])
+@pytest.mark.ignore_stream("5.10")
+def test_service_retirement_from_automate_method(request, generic_catalog_item, original_class):
+    """
+    Bugzilla:
+        1700524
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        startsin: 5.11
+        casecomponent: Automate
+        testSteps:
+            1. Create service catalog item and order
+            2. Create a writeable domain and copy ManageIQ/System/Request to this domain
+            3. Create retire_automation_service instance and set meth5 to retire_automation_service.
+            4. Create retire_automation_service method with sample code given below:
+               > service = $evm.root['service']
+               > $evm.log(:info, "create_retire_request for  service #{service}")
+               > request = $evm.execute(:create_retire_request, service)
+               > $evm.log(:info, "Create request for create_retire_request #{request}")
+            5. Execute this method using simulation
+        expectedResults:
+            1. Service provision request should be provisioned successfully
+            2.
+            3.
+            4.
+            5. Service should be retired successfully
+    """
+    # Ordering catalog item and deleting request once service has been reached to 'Finished' state
+    service_request = generic_catalog_item.appliance.rest_api.collections.service_templates.get(
+        name=generic_catalog_item.name
+    ).action.order()
+    request.addfinalizer(lambda: service_request.action.delete())
+    wait_for(lambda: service_request.request_state == "finished", fail_func=service_request.reload,
+             timeout=180, delay=10)
+
+    # Ruby code to execute create_retire_request
+    script = (
+        "service = $evm.root['service']\n"
```

Comment:
optional:
but in such cases; I would suggest to use `textwrap` `dedent`. It will help to manage script in proper way.
you can refer

https://github.com/ManageIQ/integration_tests/blob/c6913cc5bf0ff2f0c8186c5a84be0f9ea4a8e620/cfme/tests/automate/custom_button/test_service_vm_custom_button.py#L46